aiscripts/mule.lib.debug.dump_tradeoffers.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+<aiscript name="mule.lib.debug.dump_tradeoffers" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="p:\X4\Database\libraries\aiscripts.xsd" version="8">
+	<params>
+		<!-- describtive header to be included before dump of offers -->
+		<param name="headerline" default="offers"/>
+		<!-- list of possible offers -->
+		<param name="offers"/>
+		<!-- enable/disable debug messages -->
+		<param name="debugchance"/>
+		<param name="debugFileName"/>
+		<param name="debugDirName"/>
+  </params>
+  <attention min="unknown">
+    <actions>
+			<do_if value="$debugchance">
+				<!-- print headerline & count -->
+				<debug_to_file name="$debugFileName" directory="$debugDirName" text="'----- %s (%s):'.[$headerline, $offers.count]" />
+				<!-- loop over offers and print every single one-->
+				<do_for_each name="$offer" in="$offers">
+					<do_if value="$offer.owner == $offer.seller">
+						<!-- sell offers -->
+						<debug_to_file name="$debugFileName" directory="$debugDirName" text="'%1 wants to sell %2 %3 to %4 for %5 (%6)'.[$offer.seller.knownname, $offer.amount, $offer.ware, $offer.buyer, $offer.unitprice, $offer.relativeprice]" />
```

Comment:
in this line and line 33 offer.buyer either returns a memory reference or null, what did you intend for that?

Reply:
the memory reference can be converted to a name, but for null, that's probably related to the search parameters. I will have a look into it